### PR TITLE
Remove the prefix *l* in a few functions of Fold

### PR DESCRIPTION
**Pull request by adithyaov**
Fold.ltake -> Fold.takeLE
Fold.lfilter -> Fold.filter
Fold.lfilterM -> Fold.filterM
Fold.lmap -> Fold.map
Fold.lchunksOf2 -> Fold.chunksOf2
Fold.lchunksOf -> Fold.chunksOf
Fold.lsessionsOf -> Fold.intervalsOf

### CODE REVIEW

File: src/Streamly/Internal/Data/Fold.hs [hunk 1]
```diff
@@ -450,6 +434,20 @@ transform (Pipe pstep1 pstep2 pinitial) (Fold fstep finitial fextract) =
...
+-- >>> f x = if even x then Just x else Nothing
+-- >>> fld = Fold.mapMaybe f Fold.toList
+-- >>> Stream.fold fld (Stream.enumerateFromTo 1 10)
+-- [2,4,6,8]
+--
+-- /Internal/
+{-# INLINE mapMaybe #-}
+mapMaybe :: (Monad m) => (a -> Maybe b) -> Fold m b r -> Fold m a r
+mapMaybe f = map f . filter isJust . map fromJust
+
```
Comment (harendra-kumar) [inline src/Streamly/Internal/Data/Fold.hs:450]: Why was this moved here?

File: src/Streamly/Internal/Data/Fold/Types.hs [hunk 2]
```diff
@@ -644,15 +644,15 @@ instance (Monad m, Floating b) => Floating (Fold m a b) where
 -- Internal APIs
 ------------------------------------------------------------------------------
 
--- | @(lmap f fold)@ maps the function @f@ on the input of the fold.
+-- | @(map f fold)@ maps the function @f@ on the input of the fold.
```
Comment (harendra-kumar) [inline src/Streamly/Internal/Data/Fold/Types.hs:647]: We should emphatically note that this is not the same as `fmap`. `fmap` is covariant while this contravariant.

File: src/Streamly/Internal/Data/Fold.hs (hunk 1 above)
Comment (adithyaov) [reply src/Streamly/Internal/Data/Fold.hs:450]: This was in the wrong place, to begin with. This was grouped with `rmapM` and `rsequence`.

**Review by harendra-kumar — APPROVED** [review]
